src/context/TodosContext.jsx

```diff
@@ -0,0 +1,80 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable react/prop-types */
+import {
+  useState,
+  useEffect,
+  createContext,
+  useContext,
+} from 'react';
+
+import { v4 as uuidv4 } from 'uuid';
+
+const TodosContext = createContext(null);
+
+export const TodosProvider = ({ children }) => {
+  function getInitialTodos() {
+    // getting stored items
+    const temp = localStorage.getItem('todos');
+    const savedTodos = JSON.parse(temp);
+    return savedTodos || [];
+  }
+  const [todos, setTodos] = useState(getInitialTodos());
+
+  useEffect(() => {
+    // storing todos items
+    const temp = JSON.stringify(todos);
+    localStorage.setItem('todos', temp);
+  }, [todos]);
+
+  const handleChange = (id) => {
+    setTodos((prevState) => prevState.map((todo) => {
+      if (todo.id === id) {
+        return {
+          ...todo,
+          completed: !todo.completed,
+        };
+      }
+      return todo;
+    }));
+  };
+
+  const delTodo = (id) => {
+    setTodos([
+      ...todos.filter((todo) => todo.id !== id),
+    ]);
+  };
+
+  const addTodoItem = (title) => {
+    const newTodo = {
+      id: uuidv4(),
+      title,
+      completed: false,
+    };
+    setTodos([...todos, newTodo]);
+  };
+
+  const setUpdate = (updatedTitle, id) => {
+    setTodos(
+      todos.map((todo) => {
+        if (todo.id === id) {
+          todo.title = updatedTitle;
+        }
+        return todo;
+      }),
```

Comment:
[Optional] To ensure `no-param-reassign` ESLint check passes, you can do the following;
```suggestion
      todos.map((todo) => {
        if (todo.id === id) {
          return {...todo, title: updatedTitle};
        }
        return todo;
      }),
```